pymeasure/display/widgets.py

```diff
@@ -138,21 +138,42 @@ def change_y_axis(self, axis):
         for item in self.plot.items:
             if isinstance(item, ResultsCurve):
                 item.y = axis
-                item.update()
+                item.update_data()
         label, units = self.parse_axis(axis)
         self.plot.setLabel('left', label, units=units, **self.LABEL_STYLE)
         self.y_axis = axis
         self.y_axis_changed.emit(axis)
 
+class TabWidget(object):
+    """ Utility class to define default implementation for some basic methods """
+    def __init__(self, name, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
Could you comment on why you allow passing `*args` and `**kwargs` to the `__init__` of `object`; if I'm not mistaken the `object` class doesn't allow passing any arguments.

Reply:
@CasperShippers, here I followed some suggestions from https://rhettinger.wordpress.com/2011/05/26/super-considered-super/ to cope with problem of multiple inheritance and different `__init__` signature. In particular I have used the approach of https://stackoverflow.com/a/25352819 comment and https://github.com/brentpayne/learning-python/blob/master/MixInMultipleInheritance/mixin_multiple_inheritance.py
In any case, I agree with you that `object` class should receive no parameters, but possibly when using `super` for multiple inheritance passing arguments maybe "allowed".

Reply:
Ahh yes, that makes sense.